kubernetes-client/src/main/java/io/fabric8/kubernetes/client/informers/ResourceEventHandler.java

```diff
@@ -54,4 +54,12 @@
    * @param deletedFinalStateUnknown get final state of item if it is known or not.
    */
   void onDelete(T obj, boolean deletedFinalStateUnknown);
+  
+  /**
+   * If the event handler should be called in a separate thread
+   * @return
+   */
+  default boolean isCalledAsync() {
+    return true;
+  }
```

Comment:
Shall we add some example/test for the case when the user overrides this?

Reply:
This is just to prevent the proliferation of handler threads - I'm on the fence as to whether this is a good idea.  See the pr description - if you (or anyone else) would rather limit the threads dedicated to events in a simpler way, such as just 1 thread for all handlers and/or reusing the executor passed into the informerfactory for this purpose, I'd be glad to get rid of this method.

Reply:
@rohanKanojia @manusa are you comfortable with an approach that would use a single thread for all handlers instead of an isCalledAsync method?

Reply:
took a different approach of reusing the overall executor informer.